src/model/attributes/relations/MorphOne.ts

```diff
@@ -0,0 +1,100 @@
+import { Schema as NormalizrSchema } from 'normalizr'
+import { Schema } from '../../../schema/Schema'
+import { Element, Collection } from '../../../data/Data'
+import { Query } from '../../../query/Query'
+import { Model } from '../../Model'
+import { Relation, Dictionary } from './Relation'
+
+export class MorphOne extends Relation {
+  /**
+   * The field name that contains id of the parent model.
+   */
+  protected id: string
+
+  /**
+   * The field name that contains type of the parent model.
+   */
+  protected type: string
+
+  /**
+   * The local key of the model.
+   */
+  protected localKey: string
+
+  /**
+   * Create a new morph-one relation instance.
+   */
+  constructor(
+    parent: Model,
+    related: Model,
+    id: string,
+    type: string,
+    localKey: string
+  ) {
+    super(parent, related)
+    this.id = id
+    this.type = type
+    this.localKey = localKey
+  }
+
+  /**
+   * Get all related models for the relationship.
+   */
+  getRelateds(): Model[] {
+    return [this.related]
+  }
+
+  /**
+   * Define the normalizr schema for the relation.
+   */
+  define(schema: Schema): NormalizrSchema {
+    return schema.one(this.related, this.parent)
+  }
+
+  /**
+   * Attach the parent type and id to the given relation.
+   */
+  attach(record: Element, child: Element): void {
+    child[this.id] = record[this.localKey]
+    child[this.type] = this.parent.$entity()
+  }
+
+  /**
+   * Set the constraints for an eager load of the relation.
+   */
+  addEagerConstraints(query: Query, models: Collection): void {
+    query.where(this.type as any, this.parent.$entity())
+    query.whereIn(this.id as any, this.getKeys(models, this.localKey))
```

Comment:
Quick explanation of the where constraints on the relation:

The type field should match the parent's entity and the id field should match the parent's local key

Reply:
Removed `as any` as I don't think it is required for these queries.

Reply:
Nice! Actually I think this `as any` is left here because before we were trying to type hint `where` query but we're not anymore, (figured out that it's nearly impossible). So I think we can remove this one entirely for other relations as well.

We can do that in another PR though.

Reply:
Follow-up PR here: https://github.com/vuex-orm/vuex-orm-next/pull/97